src/inference/src/dev/core_impl.cpp

```diff
@@ -1127,29 +1137,62 @@ std::string ov::findPluginXML(const std::string& xmlFile) {
 
 ov::AnyMap ov::flatten_sub_properties(const std::string& device, const ov::AnyMap& properties) {
     ov::AnyMap result = properties;
-    bool isVirtualDev = device.find("AUTO") != std::string::npos || device.find("MULTI") != std::string::npos ||
-                        device.find("HETERO") != std::string::npos;
     for (auto item = result.begin(); item != result.end();) {
-        auto parsed = parseDeviceNameIntoConfig(item->first);
-        if (!item->second.is<ov::AnyMap>()) {
-            item++;
-            continue;
-        }
-        if (device == parsed._deviceName) {
-            // 1. flatten the scondary property for target device
-            for (auto&& sub_property : item->second.as<ov::AnyMap>()) {
-                // 1.1 1st level property overides 2nd level property
+        auto is_virtual_device = [&](const std::string& name) -> bool {
+            return (device.find("AUTO") != std::string::npos || device.find("MULTI") != std::string::npos ||
+                    device.find("HETERO") != std::string::npos);
+        };
+        auto update_device_property = [&](const ov::AnyMap& sub_properties) -> void {
+            for (auto&& sub_property : sub_properties) {
+                // 1st level property overrides 2nd level property
                 if (result.find(sub_property.first) != result.end())
                     continue;
                 result[sub_property.first] = sub_property.second;
             }
+        };
+        if (item->first == ov::device::properties) {
+            // 1. device properties are found
```

Comment:
```suggestion
            // 1. device properties are found in form { { "DEVICE1" : {... } }, { "DEVICE2" : {...}  } }
```
just to make comments more clear..